src/controllers/projects.controller.js

```diff
@@ -101,48 +271,49 @@ async function createProject(user, project, req, res, next) {
     return res.status(422).json({ errors: errors.mapped() });
   }
 
-  const zipper = zip();
-  const configs = project;
+  let configs = {
+    projectConfigurations: project
+  };
+  let projectId;
 
   try {
-    configs.credentials = {};
-    if (configs.toolingConfigurations.sourceControl &&
-        configs.toolingConfigurations.sourceControl.toLowerCase() === 'github') {
-      configs.credentials.github = await githubAuthService.getGithubTokenAndUsernameForUser(user);
-    }
-    if (configs.toolingConfigurations.deployment &&
-        configs.toolingConfigurations.deployment.toLowerCase() === 'heroku') {
-      configs.credentials.heroku = await herokuAuthService.getHerokuTokenAndEmailForUser(user);
-      configs.projectConfigurations.herokuAppName = configs.projectConfigurations.projectName;
-    }
-
+    configs = await updateConfigs(configs, user);
+    const newProject = await projectService.createProject(configs.projectConfigurations, user);
+    projectId = newProject.id;
     if (!fs.existsSync(`${projectLocation}`)) {
       fs.mkdirSync(`${projectLocation}`);
     }
     const filePath = `${projectLocation}/${user}`;
     if (!fs.existsSync(filePath)) {
       fs.mkdirSync(filePath);
     }
-
-    await tyr.generateBasicNodeProject(configs, filePath);
-    await tyr.generateStaticFiles(configs, filePath);
-
-    // Only do this if there is success upon creating the project
-    await projectService.createProject(configs.projectConfigurations, user);
-
-    const projectName = project.projectConfigurations.projectName;
-
-    zipper.addLocalFolder(`${filePath}/${projectName}`, `${projectName}`, undefined);
-    res.set('Content-Type', 'application/zip');
-    res.send(await zipper.toBuffer());
+    const projectPath = `${filePath}/${projectId}`;
+    if (!fs.existsSync(projectPath)) {
+      fs.mkdirSync(projectPath);
+    }
+    await tyr.generateBasicNodeProject(configs, projectPath);
+    await tyr.generateStaticFiles(configs, projectPath);
+    res.set('Content-Type', 'application/json');
+    res.send(newProject);
 
     try {
       await tyr.setUpThirdPartyTools(configs);
-      await tyr.commitToGithub(configs, filePath);
+      await tyr.commitToGithub(configs, projectPath);
+      await updateProjectTools(configs, projectId);
     } catch (error) {
       // do nothing, we already sent the res
+      console.log(error);
     }
   } catch (error) {
+    console.log(error);
+    try {
+      if (projectId) {
+        // If there was an error, and the project was created, delete it
+        await projectService.deleteProjectById(projectId, false);
+      }
+    } catch (deleteProjectError) {
```

Comment:
if we do nothing, why have a try catch?

Reply:
Because the project may still throw an error, and that may be sent to the user if it is not caught, but that would be misleading cause the first error is the real problem.